PxWeb/Code/BackgroundWorker/LongRunningService.cs

```diff
@@ -20,8 +20,8 @@ protected override async Task ExecuteAsync(CancellationToken stoppingToken)
             while (!stoppingToken.IsCancellationRequested)
             {
                 var workItem = await queue.DequeueAsync(stoppingToken);
+                string id = getControllerIdFromTask(workItem);
 
-                string id = workItem.Method.DeclaringType.FullName;
                 ResponseState state = _stateProvider.Load(id);
 
                 state.Begin();
```

Comment:
You shoukld wrap the running of the job in a try catch statement and log any exceptioons that might occure.